src/main/kotlin/Network.kt

```diff
@@ -11,14 +11,22 @@ class Network(val capacity: Int, val listOfCars: MutableList<Car>) {
     }
 
     fun analyzeNetwork() {
-        val requiredCapacity = this.requiredCapacity()
-        val carsDividedByCapacity = requiredCapacity / capacity.toDouble()
-        val chanceOfDelay = chanceOfDelay(carsDividedByCapacity)
+        val hours = 24
+        for (hour in 1..hours) {
+            val requiredCapacity = this.requiredCapacity()
+            val carsDividedByCapacity = requiredCapacity / capacity.toDouble()
+            val chanceOfDelay = chanceOfDelay(carsDividedByCapacity)
 
-        for (car in listOfCars) {
-            if (car.wantsToDrive) {
-                val delayed = computingDelay(chanceOfDelay)
-                car.isDelayed = delayed
+            for (car in listOfCars) {
+                if (car.wantsToDrive) {
+                    val delayed = computingDelay(chanceOfDelay)
+                    car.isDelayed = delayed
+                }
+                if (car.wantsToDrive && car.isDelayed == true) {
+                    car.wantsToDrive = true
+                    car.wantsToDriveAtHour.add(hour)
+                    car.isDelayedAtHour.add(hour)
```

Comment:
Yes, I think that would be best. The simulations works, after all - it just doesn't exhibit this particular behavior.